state-chain/pallets/cf-lending-pools/src/general_lending.rs

```diff
@@ -0,0 +1,1571 @@
+	// FEE_SWAP_THRESHOLD_USD in value
+	if current_block % config.fee_swap_interval_blocks.into() == 0u32.into() {
+		for pool_asset in GeneralLendingPools::<T>::iter_keys().collect::<Vec<_>>() {
+			GeneralLendingPools::<T>::mutate(pool_asset, |maybe_pool| {
+				let pool: &mut LendingPool<T> =
+					maybe_pool.as_mut().expect("Iterating over keys obtained a line above");
+
+				for (collateral_asset, fee_amount) in &mut pool.collected_fees {
+					let Ok(fee_usd_value) = usd_value_of::<T>(*collateral_asset, *fee_amount)
+					else {
+						// Don't swap yet if we can't determine asset's price
+						continue;
+					};
+
+					if fee_usd_value >= config.fee_swap_threshold_usd {
+						let fees_to_swap = core::mem::take(fee_amount);
+						let swap_request_id = initiate_swap::<T>(
+							*collateral_asset,
+							fees_to_swap,
+							pool_asset,
+							LendingSwapType::FeeSwap { pool_asset },
+							FEE_SWAP_MAX_ORACLE_SLIPPAGE,
+						);
+
+						Pallet::<T>::deposit_event(Event::LendingFeeCollectionInitiated {
+							asset: pool_asset,
+							swap_request_id,
+						});
+					}
+				}
+			})
+		}
+
+		// Additionally swapp all network fee contributions from fees:
+		for asset in PendingNetworkFees::<T>::iter_keys().collect::<Vec<_>>() {
+			PendingNetworkFees::<T>::mutate(asset, |fee_amount| {
+				// NOTE: if asset is FLIP, we shouldn't need to swap, but it should still work,
+				// and it seems easiest to not write a special case (esp if we only support
+				// boost for BTC)
+				if *fee_amount > 0 {
+					let _swap_request_id =
+						T::SwapRequestHandler::init_network_fee_swap_request(asset, *fee_amount);
+				}
+				*fee_amount = 0;
+			});
+		}
+	}
+
+	Weight::zero()
+}
+
+impl<T: Config> LendingApi for Pallet<T> {
+	type AccountId = T::AccountId;
+
+	/// Create a new loan (assigning a new loan id) provided that the account's existing collateral
+	/// plus any `extra_collateral` is sufficient. Will update the primary collateral asset if
+	/// provided.
+	#[transactional]
+	fn new_loan(
+		borrower_id: T::AccountId,
+		asset: Asset,
+		amount_to_borrow: AssetAmount,
+		primary_collateral_asset: Option<Asset>,
+		extra_collateral: BTreeMap<Asset, AssetAmount>,
+	) -> Result<LoanId, DispatchError> {
+		ensure!(
+			T::SafeMode::get().borrowing_enabled.contains(&asset),
+			Error::<T>::LoanCreationDisabled
+		);
+
+		let config = LendingConfig::<T>::get();
+
+		let loan_id = NextLoanId::<T>::get();
+		NextLoanId::<T>::set(loan_id + 1);
+
+		LoanAccounts::<T>::mutate(&borrower_id, |maybe_account| {
+			let account = Self::create_loan_account_if_empty(
+				borrower_id.clone(),
+				maybe_account,
+				primary_collateral_asset,
+			)?;
+
+			let primary_collateral_asset = account.primary_collateral_asset;
+
+			ensure!(
+				extra_collateral.contains_key(&primary_collateral_asset),
+				Error::<T>::InvalidLoanParameters
+			);
+
+			account.primary_collateral_asset = primary_collateral_asset;
```

Comment:
```
let primary_collateral_asset = account.primary_collateral_asset;
//..
account.primary_collateral_asset = primary_collateral_asset;
```

Seems redundant. Also looks like it overrides the user's provided `primary_collateral_asset`. Is this intentional?